desc/objectives/objective_funs.py

```diff
@@ -325,35 +317,34 @@ def build(self, use_jit=None, verbose=1):
         else:
             self._scalar = False
 
-        self._set_derivatives()
+        self._set_things()
+
+        # setting derivative mode and chunking.
+        errorif(
+            isposint(self._jac_chunk_size) and self._deriv_mode in ["auto", "blocked"],
+            ValueError,
+            "'jac_chunk_size' was passed into ObjectiveFunction, but the "
+            "ObjectiveFunction is not using 'batched' deriv_mode",
+        )
         sub_obj_jac_chunk_sizes_are_ints = [
             isposint(obj._jac_chunk_size) for obj in self.objectives
         ]
         errorif(
-            any(sub_obj_jac_chunk_sizes_are_ints) and self._deriv_mode != "blocked",
+            any(sub_obj_jac_chunk_sizes_are_ints) and self._deriv_mode == "batched",
```

Comment:
This is again a design choice but let's say I have a single objective,
```python
objective = ObjectiveFunction(ForceBalance(eq, jac_chunk_size))
```
This may complicate the logic but could we add something like `errorif( ..... and not len(self._objectives) == 1)` ? 

Some other edge case could be giving same `jac_chunk_size` to each objective, which should be allowed but I don't think anyone wanna use it that way, it is more laborious...